wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/ShareableDebugReport.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+import WireDesign
+import WireSyncEngine
+
+struct ShareableDebugReport: Shareable {
+
+    // MARK: - Types
+
+    typealias I = ZMConversation
+
+    // MARK: - Properties
+
+    private let logFileMetadata: ZMFileMetadata
+    private let userSession: UserSession
+
+    // MARK: - Life cycle
+
+    init(
+        logFileMetadata: ZMFileMetadata,
+        userSession: UserSession
+    ) {
+        self.logFileMetadata = logFileMetadata
+        self.userSession = userSession
+    }
+
+    // MARK: - Interface
+
+    func share<I>(to: [I]) {
```

Comment:
Out of interest why do you use `I` instead of `ZMConversation` directly? especially considering you use `ZMConversation` within the implementation. I believe the compiler will be happy with Shareable implementation even if delete `I` from this file.

Reply:
I **think** if you do the following the compiler will be happy although I might be wrong. 

```
func share<ZMConversation>(to: [ZMConversation]) {
    // implementation
}
```

Anyway it's a small thing. Feel free to close this comment.